libs/net/ches/ChesService.cs

```diff
@@ -0,0 +1,369 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using TNO.Ches.Configuration;
+using TNO.Ches.Models;
+using TNO.Core.Exceptions;
+using TNO.Core.Extensions;
+using TNO.Core.Http;
+using TNO.Core.Http.Models;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace TNO.Ches
+{
+    /// <summary>
+    /// ChesService class, provides a service for integration with Ches API services.
+    /// </summary>
+    public class ChesService : IChesService
+    {
+        #region Variables
+        private readonly ClaimsPrincipal _user;
+        private TokenModel _token = null;
+        private readonly JwtSecurityTokenHandler _tokenHandler;
+        private readonly ILogger<IChesService> _logger;
+        #endregion
+
+        #region Properties
+        protected IHttpRequestClient Client { get; }
```

Comment:
You should use the `OpenIdConnectRequestClient` instead of this `HttpRequestClient`.  It has a bunch of the functionality you've duplicated in this (i.e. RequestAccessToken).

Reply:
Let me know if I did that right, not 100% sure!